src/main/java/codeu/controller/ProfileServlet.java

```diff
@@ -0,0 +1,102 @@
+package codeu.controller;
+
+import codeu.model.store.basic.ConversationStore;
+import codeu.model.store.basic.MessageStore;
+import codeu.model.store.basic.UserStore;
+import codeu.model.data.User;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+/**
+ * Servlet class responsible for the profile page.
+ */
+public class ProfileServlet extends HttpServlet {
+
+    /**
+     * Store class that gives access to Conversations.
+     */
+    private ConversationStore conversationStore;
+
+    /**
+     * Store class that gives access to Messages.
+     */
+    private MessageStore messageStore;
+
+    /**
+     * Store class that gives access to Users.
+     */
+    private UserStore userStore;
+
+    /**
+     * Set up state for handling chat requests.
+     */
+    @Override
+    public void init() throws ServletException {
+        super.init();
+        setConversationStore(ConversationStore.getInstance());
+        setMessageStore(MessageStore.getInstance());
+        setUserStore(UserStore.getInstance());
+    }
+
+    /**
+     * Sets the ConversationStore used by this servlet. This function provides a common setup method
+     * for use by the test framework or the servlet's init() function.
+     */
+    void setConversationStore(ConversationStore conversationStore) {
+        this.conversationStore = conversationStore;
+    }
+
+    /**
+     * Sets the MessageStore used by this servlet. This function provides a common setup method for
+     * use by the test framework or the servlet's init() function.
+     */
+    void setMessageStore(MessageStore messageStore) {
+        this.messageStore = messageStore;
+    }
+
+    /**
+     * Sets the UserStore used by this servlet. This function provides a common setup method for use
+     * by the test framework or the servlet's init() function.
+     */
+    void setUserStore(UserStore userStore) {
+        this.userStore = userStore;
+    }
+
+    /**
+     * This function fires when a user navigates to a profile page. It checks if the user exists from the username of the URL requested.
+     * It then forwards to profile.jsp for rendering.
+     */
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+
+        String requestUrl = request.getRequestURI();
+        String username = requestUrl.substring("/users/".length());
```

Comment:
Hello Anita, can you explain me how do this line of code works? 
I've seen it in other servlets as well, so for example if I try to search for /users/Anita. 
Would that line of code do this? 
Okay so the request was made with Anita (length = 5), grab length characters after "/users/" and assign it to username String?

Reply:
Hi Aaron! Yes, so the length of "/users/" will always be 7, and I could have hardcoded the number in, but I think it's better to have "/users/".length() so it's easier to read? And this gets the length of 7 so it knows where the username starts so it knows to only get the substring starting from directly after the "/users/" part. So it gets requestUrl.substring(7) which will be the username.